ouroboros-consensus/docs/VersioningSchemeDecision.md

```diff
@@ -0,0 +1,286 @@
+
+To cut a release from a commit COMMIT on `main` that declares `A.B.C`, announce COMMIT.
+Also immediately update the `main` version to `A.B.(C+1)`.
+
+Cons:
+
+- `main` versions and release versions are not distinguished.
+
+- Some `main` commits will declare version `A.B.(C+1)` even if that version is never officially released or is created only later by backporting _different_ (patch) commits to a previous release branch.
+
+## Proposal Parity
+
+Each `main` version `A.B.C` has an odd `B`.
+Each release version `A.B.C` has an even `B`.
+This is similar to the GHC Team's scheme.
+
+PRs do not alter the `main` version.
+
+To cut a release from a commit COMMIT on `main` that declares version `A.B` (where `B` is necessarily odd), announce a new non-`main` commit that extends COMMIT merely to declare version `X.Y = next(A.B)` (where `Y` is necessarily even).
+Also immediately update the `main` version to `X.(Y+1)`.
+
+Cons:
+
+- Commits on `main` that only include patches since the previous release would still spuriously include the `Y+1` increment in the `B` dimension.
+
+## Proposal NonZero
+
+Each `main` version is degenerate; eg it is always version `0`.
+Each release version is the usual `A.B.C`.
+
+PRs do not alter the `main` version.
+
+To cut a release from a commit COMMIT on `main` (that necessarily declares version `0`), announce a new non-`main` commit that extends COMMIT merely to declare the version to be `next(A.B).0`, or `A.B.(C+1)` depending on what has changed on `main` since the previous release.
+
+Note that the degenerate versions could carry information.
+EG they could be just a single number.
+The only requirement is that they are inherently distinguished from release versions.
+
+Cons:
+
+- `main` versions would always be less than some older released version, which could cause confusion (among people and/or tools).
+
+- Cutting a release requires assessing all the changes on `main` since the last release.
+
+## Proposal Dimension
+
+FYI
+
+```
+Prelude Data.Version> makeVersion [1,2] < makeVersion [1,2,0]
+True
+```
+
+Each `main` version has only two dimensions: `A.B`.
+Each release version has at least three `A.B.C`, where `C` can be `0`.
+
+PRs do not alter the `main` version.
+
+To cut a release from a commit COMMIT on `main` that declares version `A.B`, announce a new non-`main` commit that extends COMMIT merely to declare version `A.B.0`.
+Also immediately update the `main` version to `next(A.B)`.
+
+Pros:
+
+- It enforces all desired invariants.
+
+- It's a very mechanical state machine, easy to execute and also easy to immediately recognize which state it's in.
+
+Cons:
+
+- The multi-sorted state transition system prevents any explanation from being comparatively small.
+
+- This scheme is certainly not already well-established!
+
+- It incurs spurious increments of `A.B` when the only differences between two releases were patch PRs.
```

Comment:
I don't understand why this has to be the case. Can you not, in this proposal, treat `C` the same as `A` and `B`, and replace `A.B` with `A.B.C` across the board?